Source: John <johnz@posteo.net>
Upsteam: No
Subject: Musl returns /lib/libQt5Core.so.5 for Dl_info.dli_fname. 
GLIBC's dl behaves differently and therefore the path is prefixed with /usr.
This broke e.g. qt5-webengine on musl.
---
diff --git qtbasesrc/corelib/global/qlibraryinfo.cpp qtbasesrc/corelib/global/qlibraryinfo.cpp
index 8bcf67e73d..28d7355310 100644
--- qtbase/src/corelib/global/qlibraryinfo.cpp
+++ qtbase/src/corelib/global/qlibraryinfo.cpp
@@ -558,6 +558,12 @@ static QString getRelocatablePrefix()
     int result = dladdr(reinterpret_cast<void *>(&QLibraryInfo::isDebugBuild), &info);
     if (result > 0 && info.dli_fname)
         prefixPath = prefixFromQtCoreLibraryHelper(QString::fromLatin1(info.dli_fname));
+#ifndef __GLIBC__
+    QDir prefixDir(prefixPath);
+    if (prefixDir.isRoot()) {
+        prefixPath = "/usr";
+    }
+#endif
 #elif defined(Q_OS_WIN)
     HMODULE hModule = getWindowsModuleHandle();
     const int kBufferSize = 4096;
